components/micro-gateway-core/src/main/ballerina/src/gateway/handlers/jwt_auth_handler.bal

```diff
@@ -113,12 +114,16 @@ public type JWTAuthHandler object {
 
 # Check whether backendJwt claim is in the payload and set the header if avaialable.
 #
+# + jwtValidatorConfig - jwtValidatorConfig to access the jwtCache object
 # + credential - Credential
 # + req - The `Request` instance.
 # + issuer - The jwt issuer who issued the token and comes in the iss claim.
 # + return - Returns boolean based on backend jwt setting.
-public function setBackendJwtHeader(string credential, http:Request req, string? issuer) returns @tainted boolean {
-    (jwt:JwtPayload | error) payload = getDecodedJWTPayload(credential, issuer);
+public function setBackendJwtHeader(jwt:JwtValidatorConfig jwtValidatorConfig,
+                                    string credential,
+                                    http:Request req)
+                                    returns @tainted boolean {
+    (jwt:JwtPayload | error) payload = getDecodedJWTPayload(jwtValidatorConfig, credential);
```

Comment:
Use the principal instead. Otherwise, if the cache is disabled, these all locations will decode the JWT repeatedly.

Reply:
updated with https://github.com/wso2/product-microgateway/pull/1345/commits/07f4cea95eef0b1a7edceb32af927a6dcc74ade7